packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -70,31 +75,47 @@ export class WaferMap extends FoundationElement {
 
     private renderQueued = false;
     private dataManager: DataManager | undefined;
-
+    private renderer: RenderingModule | undefined;
+    private resizeObserver: ResizeObserver | undefined;
     public override connectedCallback(): void {
         super.connectedCallback();
+        this.resizeObserver = new ResizeObserver(entries => {
+            const entry = entries[0];
+            if (entry === undefined) return;
+            const { height, width } = entry.contentRect;
+            this.canvasSideLength = Math.min(height, width);
+        });
+        this.resizeObserver.observe(this);
         this.queueRender();
     }
 
+    public override disconnectedCallback(): void {
+        this.resizeObserver?.unobserve(this);
```

Comment:
Make sure to call super.disconnectedCallback(); first since this is an override

Reply:
applied